packages/data/addon/builder/requestV2.ts

```diff
@@ -0,0 +1,176 @@
+   * @param alias - Optoinal Display name for the field
+   * @param cid - column id, generated if not given
+   */
+  column(type: ColumnType, field: string, parameters: Parameters, alias?: string, cid?: string): RequestV2Builder {
+    this.request.columns.push({
+      type,
+      field,
+      parameters,
+      ...(alias ? { alias } : null),
+      cid: cid ?? nanoid(10)
+    });
+    return this;
+  }
+
+  /**
+   * Convenience for adding a metric column
+   * @param metricName Name of metric
+   * @param parameters Dictionary of parameters
+   */
+  metric(metricName: string, parameters: Parameters = {}): RequestV2Builder {
+    this.column('metric', metricName, parameters);
+    return this;
+  }
+
+  /**
+   * Convenience for adding a dimension column
+   * @param dimensionName Name of dimension
+   * @param parameters Dictionary of parameters
+   */
+  dimension(dimensionName: string, parameters: Parameters = {}): RequestV2Builder {
+    this.column('dimension', dimensionName, parameters);
+    return this;
+  }
+
+  /**
+   * Adds a filter to the request
+   * @param type metric, dimension, or timeDimension
+   * @param field field name
+   * @param parameters dictionary of parameters
+   * @param operator operator
+   * @param values array of values for the filter
+   */
+  filter(
+    type: ColumnType,
+    field: string,
+    parameters: Parameters,
+    operator: FilterOperator,
+    values: (string | number | boolean)[]
+  ): RequestV2Builder {
+    this.request.filters.push({
+      type,
+      field,
+      parameters,
+      operator,
+      values
+    });
+    return this;
+  }
+
+  /**
+   * Adds a row limit to request
+   * @param limit number of max rows
+   */
+  limit(limit: number): RequestV2Builder {
+    this.request.limit = limit;
+    return this;
+  }
+
+  /**
+   * Adds sort to the request
+   * @param type metric,dimension or timeDimension
+   * @param field field name to sort by
+   * @param parameters Dictionary of parameters
+   * @param direction asc/desc
+   */
+  sort(type: ColumnType, field: string, parameters: Parameters, direction: SortDirection): RequestV2Builder {
+    this.request.sorts.push({
+      type,
+      field,
+      parameters,
+      direction
+    });
+    return this;
+  }
+
+  /**
+   * Adds table name to request
+   * @param name name of the table
+   */
+  table(name: string): RequestV2Builder {
+    this.request.table = name;
+    return this;
+  }
+
+  /**
+   * Allows specifies the datasource
+   * @param dataSource name of the datasource
+   */
+  dataSource(dataSource: string): RequestV2Builder {
+    this.request.dataSource = dataSource;
+    return this;
+  }
+
+  /**
+   * Convenience method for adding an interval filter
+   * @param grain 'day','week','month' etc... whatever your datasource supports
+   * @param start Period, datestring or macro
+   * @param end Period, datestring or macro
+   */
+  interval(grain: string, start: string, end: string): RequestV2Builder {
```

Comment:
We are moving away from the concept of interval, should we rename this to something like this:

```suggestion
  addTimeRangeFilter(grain: Grain, start: string, end: string): RequestV2Builder {
```


Also, we can use the `Grain` type from packages `navi-data/addon/utils/date.ts`